### DISCUSSION TITLE
Gordon cannon be used in test module

### DISCUSSION THREAD

**Discussion by christianbrennstuhlbmw** [Banno/Gordon#119] [Ideas]
Gordon can only be used if `application`, `library` or `dynamic-feature` has been applied before.

If Gordon is now used in a test module, where plugin `com.android.test` is applied before, it runs into [this error](https://github.com/Banno/Gordon/blob/main/gordon-plugin/src/main/kotlin/com/banno/gordon/GordonPlugin.kt#L27).

As test modules might be pretty common in bigger projects, I think it should be possible to use Gordon.
It might be quick win, as afaik test modules are similar to library modules, so maybe it is just copying [these code lines](https://github.com/Banno/Gordon/blob/main/gordon-plugin/src/main/kotlin/com/banno/gordon/GordonPlugin.kt#L147-L165).

**Comment by christianbrennstuhlbmw** [comment]
@joshschriever 
WDYT? :)

**Comment by joshschriever** [comment]
Unfortunately it won't be that simple. Gordon still uses `com.android.build.gradle.TestedExtension` to get `testInstrumentationRunnerArguments`. I haven't found any way to get that same information using the newer AGP APIs. And `TestExtension` (which is what the `com.android.test` plugin uses) does not implement that interface.
